### PR TITLE
feat: add a featured card to the hero section in the author page

### LINKED ISSUE #9952: Show external posts on new author pages
Related issues https://github.com/GoogleChrome/webdev-infra/issues/71 and #9932

Please update the layout of the feeds collection section of the author profile to be able to display their “internal” and “external” posts.

The design note in the Figma:

> We essentially have “internal” and “external” posts. Internal would be posts made directly to the feed (e.g. announcements, or even sharing external links, which are still posts made by the author). Then, we have posts which are pulled from external feeds (e.g. personal blogs, d.c.c. pages).

Also, add the filter component to allow users to filter the posts by these options:
- by source (e.g. web.dev., developer.chrome.com, external)
- by type (e.g. announcement, article, patterns, RSS feed)
- by credit (e.g. author, editor)

Lastly, updated hero section with option to add a “featured” card.

### PR DESCRIPTION
**Pull request by mamieorine**
Fixes #9952 

Updated the hero section with option to add a “featured” card.  

Screenshot 2023-05-02 at 11 34 11

The featuredPost properties available:

- `title` - the title of the card, like a event's name or a post's title.
- `summary` - a description or short summary.
- `thumbnail` - the thumbnail of the card.
- `eyebrowText` - the keyword to introduce the type of card.
- `eyebrowIcon` - the icon to introduce the type of card, such as mortarboard, pattern, podcast, blog, news, etc. The default value is featured.
- `url` - the url for the page the card is previewing.
- `alt` - an alt text for image. If no thumbnail is provided, this can be null.
- `theme` - the theme colours of the card. Theme colours available - tertiary, quaternary, pink, dark, and blue. The default theme is quaternary.

If an author wants to set an internal post to be a featured post, copy the post URL and use it as the `url` value for the `featuredPost` key in your profile in the [authorsData.json](https://github.com/GoogleChrome/web.dev/blob/main/src/site/_data/authorsData.json). It will automatically fetch the all data from the post URL, but also allow an author to override the detail, such as title, eyebrow, thumbnail, alt, and summary.

Example usage:

```
"paulkinlan": {
  "homepage": "https://paul.kinlan.me/",
  "twitter": "paul_kinlan",
  "github": "PaulKinlan",
  "glitch": "PaulKinlan",
  "dcc": "paulkinlan",
  "mastodon": "https://status.kinlan.me/@paul",
  "linkedin": "https://uk.linkedin.com/in/paulkinlan",
  "image": "image/T4FyVKpzu4WKF1kBNvXepbi08t52/0O1ZGr2P0l9oTKabyUK5.jpeg",
  "featuredPost": {
   "url": "/googleio22-recap/",
   "eyebrowText": "Learn",
   "eyebrowIcon": "mortarboard",
   "title": "A simple TODO list using HTML5 WebDatabases",
   "thumbnail": "image/cGQxYFGJrUUaUZyWhyt9yo5gHhs1/9WSNd3mdbXACF19ELKJ1.png",
   "summary": "This tool by Josh W Comeau makes it super simple to create nice looking gradients.",
   "alt": "HTML5 text on the black background",
   "theme": "quaternary"
  }
 }
```

### CODE REVIEW

**Review by Karl2013 — COMMENTED** [review]
@twifkak

File: src/site/_filters/i18n.js [hunk 1]
```diff
@@ -74,9 +74,11 @@ const i18n = (path, locale = defaultLocale) => {
       return out;
     }
   } catch (err) {
-    // ignore, throw below
+    // ignore, return an empty string in the case of no i18n result.
   }
-  throw new Error(`Could not find i18n result for: ${path}`);
+
+  console.warn('Could not find i18n result');
```
Comment (matthiasrohmer) [inline src/site/_filters/i18n.js:80]: I see why that change is necessary, but it's not ideal... building the site no prints 816x 'Could not find i18n result', which might be confusing for users... also not throwing anymore can easily introduce translation errors, which would only make the build process render the warning 817 times 😄

The new i18n implementation in webdev-infra falls back to the `en` translation instead, and only errors if that doesn't exist too. I think we should do the same here as an interim solution once we switch to the new i18n filter.

**Review by matthiasrohmer — COMMENTED** [review]
- Hovering the featured card yields weird effects

image

- I think for mobile, the featured card width should follow the widths of the grid below, see:
image
image

File: src/scss/pages/_collection.scss [hunk 2]
```diff
@@ -191,72 +191,72 @@
...
-  .bg-dark,
-  .bg-blue  {
-    svg {
-        fill: get-color('shades-light-bright');
-    }
-  
-    p, span, h4 {
-        color: get-color('shades-light-bright');
-    }
+}
```
[the remark this answers is not in the document]
Comment (mamieorine) [reply src/scss/pages/_collection.scss:194]: I want the featured card on the author page to be able to access the theme for the cards in the collection page, so I moved all theme setting out of the `.collection` parent.

**Pull request closed without merging**